src/OpenTracing/ISpan.cs

```diff
@@ -1,164 +1,177 @@
-﻿using System;
-using System.Collections.Generic;
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="ISpan.cs">
+//   Copyright 2017-2018 The OpenTracing Authors
+//   
+//   Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+//   in compliance with the License. You may obtain a copy of the License at
+//   
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+//   Unless required by applicable law or agreed to in writing, software distributed under the License
+//   is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+//   or implied. See the License for the specific language governing permissions and limitations under
+//   the License.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
 
 namespace OpenTracing
 {
+    using System;
+    using System.Collections.Generic;
+
     /// <summary>
-    /// Represents an in-flight span in the opentracing system.
-    /// Spans are created by the <see cref="ITracer"/> interface.
+    ///     Represents the OpenTracing specification's Span contract.
+    ///     <seealso cref="IScope"/>
+    ///     <seealso cref="IScopeManager"/>
+    ///     <seealso cref="ISpanBuilder.StartManual"/>
+    ///     <seealso cref="ISpanBuilder.StartActive()"/>
     /// </summary>
-    public interface ISpan : IDisposable
+    public interface ISpan
     {
         /// <summary>
-        /// Returns the <see cref="ISpanContext"/> that encapsulates span state that should propagate across process boundaries.
-        /// Note that the return value of <see cref="Context"/> is still valid after a call to <see cref="Finish()"/>, as is
-        /// a call to <see cref="Context"/> after a call to <see cref="Finish()"/>.
+        ///     Retrieve the associated SpanContext.
+        ///     This may be called at any time, including after calls to <see cref="ISpan.Finish()" />.
         /// </summary>
-        ISpanContext Context { get; }
+        /// <returns>The SpanContext that encapsulates Span state that sould propagate across process boundaries.</returns>
+        ISpanContext Context();
```

Comment:
This should definitely stay a property

Reply:
Is i t generally assumed that the ISpan will cache the value of it's ISpanContext? If the operation will potentially have allocations/processing it's best to use a method since developers are more familiar with expecting such things from a method (whereas a property they expect to work [more or less] like a field and be very quick and not necessary to cache themselves)

Reply:
hmm... I'm not sure TBH but I think it's usually just a reference. I'll look into usages tomorrow.

Reply:
My implementation appears to use a reference. Method would be safer as far as an API w/o implementations goes as it supports both and leaves the caching decision up to the caller, but if you still feel property is the way to go I am willing to change.